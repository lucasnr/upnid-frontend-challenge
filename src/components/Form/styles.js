import styled from 'styled-components';

export const Container = styled.form`
	align-items: center;
	display: flex;
	margin-bottom: 1.5rem;
	width: 20rem;

	input {
		border: none;
		border-radius: 0.125rem;
		font-size: 0.75rem;
		flex: 1;
		margin-right: 0.75rem;
		width: 100%;
		padding: 0.375rem 1rem;
	}

	button {
		background-color: #49e;
		border: none;
		border-radius: 0.125rem;
		color: #fff;
		font-size: 0.75rem;
		padding: 0.375rem 1rem;
		text-transform: uppercase;

		span {
			display: none;
		}
	}

	@media (min-width: 576px) {
		width: 30rem;

		input,
		button {
			font-size: 1rem;

			span {
				display: inline;
			}
		}
	}
`;
