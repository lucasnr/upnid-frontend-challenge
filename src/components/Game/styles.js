import styled, { keyframes } from 'styled-components';

import bg from '../../assets/img/bg.gif';
import bgPaused from '../../assets/img/bg-pause.png';
import car from '../../assets/img/car.png';
import obstacle from '../../assets/img/obstacle.png';

export const Container = styled.main`
	background-image: url(${(props) => (props.paused ? bgPaused : bg)});
	background-position: center;
	background-size: cover;
	border-radius: 0.25rem;
	height: 20rem;
	overflow: hidden;
	position: relative;
	width: 20rem;

	@media (min-width: 576px) {
		height: 30rem;
		width: 30rem;
	}
`;

export const Car = styled.img.attrs((props) => ({
	alt: 'Car icon',
	style: {
		left: `${props.left}%`,
	},
	src: car,
}))`
	bottom: 0px;
	height: 6rem;
	position: absolute;
	transition: left 300ms ease;
	width: 6rem;

	@media (min-width: 576px) {
		height: 9rem;
		width: 9rem;
	}
`;

const obstacleAnimation = (props) => keyframes`
	0% {
		height: .625rem;
		top: 45%;
		width: 1rem;
	}

	100% {
		height: 7.25rem;
		left: ${props.to}%;
		top: 100%;
		width: 8rem;
	}
`;

export const Obstacle = styled.img.attrs((props) => ({
	alt: 'Obstacle icon',
	style: {
		left: `${props.initial || 45}%`,
	},
	src: obstacle,
}))`
	animation: ${(props) => obstacleAnimation(props)} 2s linear forwards;
	position: absolute;
	transition: left 300ms ease;
`;

export const Paused = styled.span.attrs((props) => ({
	children: 'Paused',
}))`
	background-color: rgba(0, 0, 0, 0.75);
	border-radius: 0.25rem;
	color: #fff;
	font-size: 1.5rem;
	font-weight: bold;
	left: 50%;
	padding: 1rem 2rem;
	position: absolute;
	top: 50%;
	text-transform: uppercase;
	transform: translate(-50%, -50%);
`;
