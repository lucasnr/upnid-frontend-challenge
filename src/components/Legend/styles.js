import styled from 'styled-components';

export const Container = styled.section`
	margin-top: 1rem;
	width: 20rem;

	@media (min-width: 576px) {
		width: 30rem;
	}
`;

export const Name = styled.h2`
	font-size: 1rem;
	font-weight: normal;
	text-align: center;
`;

export const Controls = styled.div`
	display: grid;
	grid-template-columns: repeat(2, 1fr);
	margin-top: 1rem;

	& > * {
		font-size: 0.75rem;

		&:nth-child(even) {
			text-align: right;
		}
	}
`;
