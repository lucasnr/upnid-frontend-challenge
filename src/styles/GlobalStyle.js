import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
	* {
		box-sizing: border-box;
		margin: 0px;
		outline: none;
		padding: 0px;
	}

	ul {
		list-style: none;
	}

	a {
		color: inherit;
		text-decoration: none;
	}

	body {
		background-color: #efefef;
	}

	body, input, button {
		font-family: 'Poppins', sans-serif;
	}

	button {
		cursor: pointer;
	}

	#root {
		align-items: center;
		display: flex;
		justify-content: center;
		min-height: 100vh;
	}
`;

export default GlobalStyle;
