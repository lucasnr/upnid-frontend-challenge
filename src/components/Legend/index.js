import React from 'react';
import { isMobile } from 'react-device-detect';

import { Container, Name, Controls } from './styles';

export default function Legend({ name }) {
	return (
		<Container>
			<Name>{name || 'Jogador sem nome'}</Name>
			<Controls>
				{isMobile ? (
					<>
						<h5>Deslizar para esquerda</h5>
						<h6>Ir para a esquerda</h6>
						<h5>Deslizar para direita</h5>
						<h6>Ir para a direita</h6>
					</>
				) : (
					<>
						<h5>A</h5>
						<h6>Ir para o lado esquerdo</h6>
						<h5>S</h5>
						<h6>Ir para o centro</h6>
						<h5>D</h5>
						<h6>Ir para o lado direito</h6>
						<h5>Seta esquerda</h5>
						<h6>Ir para a esquerda</h6>
						<h5>Seta direita</h5>
						<h6>Ir para a direita</h6>
						<h5>ESC</h5>
						<h6>Pausar jogo</h6>
					</>
				)}
			</Controls>
		</Container>
	);
}
