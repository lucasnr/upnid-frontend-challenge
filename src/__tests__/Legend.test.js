import React from 'react';
import { shallow } from 'enzyme';

import Legend from '../components/Legend';
import { Name } from '../components/Legend/styles';

describe('Testing legend component', () => {
	it('renders without crashing and matches snapshot', () => {
		const wrapper = shallow(<Legend />);

		expect(wrapper).toMatchSnapshot();
	});

	it('renders default value without prop value', () => {
		const defaultValue = 'Jogador sem nome';
		const wrapper = shallow(<Legend />);

		expect(wrapper.find(Name).text()).toBe(defaultValue);
	});

	it('renders prop value', () => {
		const value = 'Some value that should be rendered';
		const wrapper = shallow(<Legend name={value} />);

		expect(wrapper.find(Name).text()).toBe(value);
	});
});
