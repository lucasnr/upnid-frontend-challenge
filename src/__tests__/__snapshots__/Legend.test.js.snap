// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing legend component renders without crashing and matches snapshot 1`] = `
<styled.section>
  <styled.h2>
    Jogador sem nome
  </styled.h2>
  <styled.div>
    <h5>
      A
    </h5>
    <h6>
      Ir para o lado esquerdo
    </h6>
    <h5>
      S
    </h5>
    <h6>
      Ir para o centro
    </h6>
    <h5>
      D
    </h5>
    <h6>
      Ir para o lado direito
    </h6>
    <h5>
      Seta esquerda
    </h5>
    <h6>
      Ir para a esquerda
    </h6>
    <h5>
      Seta direita
    </h5>
    <h6>
      Ir para a direita
    </h6>
    <h5>
      ESC
    </h5>
    <h6>
      Pausar jogo
    </h6>
  </styled.div>
</styled.section>
`;
